stator/models.py

```diff
@@ -253,6 +253,25 @@ def transition_perform(self, state: State | str):
 
     atransition_perform = sync_to_async(transition_perform)
 
+    def set_transition_state(self, state: State | str):
```

Comment:
Can you rename this to `transition_set_state` so the name is consistent with a leading `transition`?

It would be nice not to duplicate this logic, but I see why you've done it (to save the extra UPDATE).